app/oauth.go

```diff
@@ -791,35 +823,36 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Authorization", "Bearer "+ar.AccessToken)
 
-	if resp, err := a.HTTPService.MakeClient(true).Do(req); err != nil {
-		return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.service.app_error", map[string]interface{}{"Service": service}, err.Error(), http.StatusInternalServerError)
-	} else {
-		bodyBytes, _ = ioutil.ReadAll(resp.Body)
-		if resp.StatusCode != http.StatusOK {
-			bodyString := string(bodyBytes)
-			mlog.Error("Error getting OAuth user: " + bodyString)
-			if service == model.SERVICE_GITLAB && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
-				return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "oauth.gitlab.tos.error", nil, "", http.StatusBadRequest)
-			}
+	resp, serviceErr = a.HTTPService.MakeClient(true).Do(req)
+	if serviceErr != nil {
+		return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.service.app_error", map[string]interface{}{"Service": service}, serviceErr.Error(), http.StatusInternalServerError)
+	}
+
+	bodyBytes, _ = ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		bodyString := string(bodyBytes)
+		mlog.Error("Error getting OAuth user: " + bodyString)
+		if service == model.SERVICE_GITLAB && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
+			return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "oauth.gitlab.tos.error", nil, "", http.StatusBadRequest)
 		}
-		resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
-		return resp.Body, teamId, stateProps, nil
 	}
 
+	resp.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+	return resp.Body, teamId, stateProps, nil
 }
 
 func (a *App) SwitchEmailToOAuth(w http.ResponseWriter, r *http.Request, email, password, code, service string) (string, *model.AppError) {
 	if a.License() != nil && !*a.Config().ServiceSettings.ExperimentalEnableAuthenticationTransfer {
 		return "", model.NewAppError("emailToOAuth", "api.user.email_to_oauth.not_available.app_error", nil, "", http.StatusForbidden)
 	}
 
-	var user *model.User
-	var err *model.AppError
-	if user, err = a.GetUserByEmail(email); err != nil {
+	user, err := a.GetUserByEmail(email)
+	if err != nil {
 		return "", err
 	}
 
-	if err := a.CheckPasswordAndAllCriteria(user, password, code); err != nil {
+	err = a.CheckPasswordAndAllCriteria(user, password, code)
```

Comment:
Don't lift this out of the `if` as there's no `else` block that we can refactor in this case.